### PR TITLE
Convert process setting map to a slice

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/384

## Type of change

*Please select one of the options below.*

- Other

## Discussion

We can remove the process setting in the next CRD release (v1beta3).

## Testing

Unit tests (e2e tests will be running).

## Documentation

-

## Follow-up

-

### CODE REVIEW

File: internal/deprecations.go [hunk 1]
```diff
@@ -119,17 +141,39 @@ func ensureImageConfigPresent(imageConfigs *[]fdbv1beta2.ImageConfig, expected f
...
+		spec.ProcessesConfigs[settingIdx] = generalConfig
+	} else {
+		spec.ProcessesConfigs = append(spec.ProcessesConfigs, generalConfig)
+	}
+
+	//generalSettings := spec.Processes[fdbv1beta2.ProcessClassGeneral]
+	//if generalSettings.PodTemplate == nil {
+	//	generalSettings.PodTemplate = &corev1.PodTemplateSpec{}
+	//}
+	//spec.Processes[fdbv1beta2.ProcessClassGeneral] = generalSettings
```
Comment (johscheuer) [inline internal/deprecations.go:176]: This code must be removed.

File: api/v1beta2/foundationdbcluster_types.go [hunk 2]
```diff
@@ -219,6 +223,15 @@ type FoundationDBClusterSpec struct {
 	UseUnifiedImage *bool `json:"useUnifiedImage,omitempty"`
 }
 
+// ProcessesConfig represents the configuration for a processes. This will be used as a template for all processes
+// of with the specified process class.
+type ProcessesConfig struct {
+	// Class the process class that is affected by this configuration.
+	Class ProcessClass `json:"class,omitempty"`
```
Comment (brownleej) [inline api/v1beta2/foundationdbcluster_types.go:230]: I think this is going to introduce a new string field in the CRD with no max-length. Maybe we should take this opportunity to put a max-length annotation on the ProcessClass type.

File: internal/replacements/replacements_test.go [hunk 3]
```diff
@@ -501,14 +527,20 @@ var _ = Describe("replace_misconfigured_pods", func() {
...
+						setting.PodTemplate.Spec.Containers = append(setting.PodTemplate.Spec.Containers,
 							corev1.Container{
 								Resources: corev1.ResourceRequirements{
 									Requests: corev1.ResourceList{
 										corev1.ResourceCPU: newCPU,
 									},
 								},
 							})
+
+						cluster.UpdateBareProcessSettings(fdbv1beta2.ProcessClassGeneral, setting)
```
Comment (brownleej) [inline internal/replacements/replacements_test.go:543]: If we're repeating this fetch-and-update pattern multiple times, could we can encapsulate it in a function that takes a callback?

**Review by brownleej — CHANGES_REQUESTED** [review]

File: api/v1beta2/foundationdbcluster_types.go (hunk 2 above)
Comment (johscheuer) [reply api/v1beta2/foundationdbcluster_types.go:230]: Good point 👍

**Review by johscheuer — COMMENTED** [review]
Pausing this effort here since the CRD size is exploding.

**Pull request closed without merging**